### ISSUE TITLE
Make electric-quote-local-mode work

### ISSUE THREAD

**Issue by Chris00** [djcb/mu#997]
At the moment, `electric-quote-local-mode` does not work (outside "comments" seen as lines starting with `>` or `|`).  This is due to `comment-use-syntax` left to `undecided`.  As the documentation for this variable says *Major modes should set this variable*, I believe `mu4e-compose` should set to to `nil` (so `electric-quote-local-mode` works anywhere).  If there is an agreement, I can submit a PR.

**Comment by djcb** [comment]
It would be useful, but setting it to `nil` would not be enough I think; quote local mode should be neutralized for the message-headers, as well as in any of the special parts like attachment-blobs.

Can you make a PR for that?

**Comment by Chris00** [comment]
Why disable it in the headers?  `"` is not affected, only `` ` `` and `'` and these may be useful for the `Subject:`.

**Comment by Chris00** [comment]
Also, electric quote mode works by adding a function to `'post-self-insert-hook` so it only changes quotes that one types, not those present in attached files, pasted text,...

**Comment by djcb** [comment]
I wouldn't want it in headers, not convinced non-ascii will work there in all cases (and think of "Resend"). It certainly won't work in the mime-meta blocks. If not too hard, we probably also want to exclude it from code blocks, ie. 
```
--8<---------------cut here---------------start------------->8---
   code
--8<---------------cut here---------------end--------------->8---
```
Can you add that? I think all it involves setting a few of the `comment-` regexps.

Apart from that, looks useful, thanks!

**Comment by Chris00** [comment]
I checked the code of `electric-quote-post-self-insert-function` and when `comment-use-syntax` is `nil`, it basically only cares about `` ` `` or `'` being pressed.  It it is `t` (and because `comment-start` is defined), it tries to see whether the point is in a comment (quoted text for emails) or a string to enable the conversion.  I checked org-mode—which sets `comment-use-syntax` to `nil`—but it does not "protect" quotes for code blocks (one has to type `C-q '` to have `'` instead of `’`—or (temporarily) disable electric quote.

Note that `electric-quote-local-mode` does not affect `"`.
